### DISCUSSION TITLE
(Feature request) How do I reference a local image instead of inlaying it?

### DISCUSSION THREAD

**Discussion by Joackk** [TriliumNext/Trilium#7356] [Q&A] [answered]
### Describe feature

I have a lot of images in my notes, and embedding them would really take up the size of the notes. Can I store these images somewhere locally and then reference them in my notes? How do I do this?
The web reference is a direct copy of the image, but how do I refer to it locally?

**Comment by XXXJetfireXXX** [comment]
That is actually **the only way** how image notes work in Trilium as of now. When you paste an image inside a text note - an image type child note is being created and a reference to that note gets pasted into the text note. You can go to the image note (by clicking on the image reference for example) and "copy reference to clipboard", which you can then paste anywhere you like as well. Image notes that are referenced in their parent text note are hidden by default, but you can change this here:
![image](https://github.com/zadam/trilium/assets/43646178/52bc3f21-843a-4941-8c1f-73a440173091)

**Comment by Joackk** [comment]
> 这实际上是目前图像注释在Trilium中工作**的唯一方式**。将图像粘贴到文本注释中时，将创建图像类型的子注释，并将对该注释的引用粘贴到文本注释中。您可以转到图像注释（例如，通过单击图像引用）和“将引用复制到剪贴板”，然后您也可以将其粘贴到您喜欢的任何位置。默认情况下，在其父文本注释中引用的图像注释处于隐藏状态，但您可以在此处进行更改： ![image](https://user-images.githubusercontent.com/43646178/237942489-52bc3f21-843a-4941-8c1f-73a440173091.png)

I mean, instead of embedding the image into the database, it is similar to the Markdown format, where the picture is stored locally, but can they be referenced through the path? I am now going to install a drawing bed in wsl to call，Here I clicked on him and then copied the reference to the clipboard just copied the image, I was trying to modify its location so that he was not stored in the database

**Comment by zadam** [comment]
Trilium by default attempts to download all images locally so they are available offline.

You can disable this in options:

![image](https://github.com/zadam/trilium/assets/617641/44a4967d-fbc9-40c2-b41b-43c4d261f48f)

If you disable this and paste a image reference, then only the reference will be used.

**Comment by Joackk** [comment]
> Trilium by default attempts to download all images locally so they are available offline.
> 
> You can disable this in options:
> 
> ![image](https://user-images.githubusercontent.com/617641/238136533-44a4967d-fbc9-40c2-b41b-43c4d261f48f.png)
> 
> If you disable this and paste a image reference, then only the reference will be used.

That means as long as this option is turned off, then when I copy the local image in, it's not set into the database, but just like a Markdown reference image? Thank you very much for your answer, tr is the best note taking software I have ever used!

**Comment by zadam** [comment]
> That means as long as this option is turned off, then when I copy the local image in, it's not set into the database, but just like a Markdown reference image?

Yes.

But it depends on what is in the clipboard. In some cases the clipboard contains the actual bitmap and the image URL, in such a case the image is still stored in the database. But normally when you copy from e.g. browser, only the reference is in the clipboard and then it won't be stored in DB.

**Comment by Joackk** [comment]
> > That means as long as this option is turned off, then when I copy the local image in, it's not set into the database, but just like a Markdown reference image?
> 
> Yes.
> 
> But it depends on what is in the clipboard. In some cases the clipboard contains the actual bitmap and the image URL, in such a case the image is still stored in the database. But normally when you copy from e.g. browser, only the reference is in the clipboard and then it won't be stored in DB.

Got it. Thank you very much.

**Comment by eliandoran** [comment, marked as answer]
Images are now saved as attachments, so they will no longer appear as part of the tree.
